### PR TITLE
[API] Order confirmation email sending

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

This PR introduce mailing refactor, which should be continued in the future. Bigger description and ADR in progress.

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Resources/views/Email/Blocks/OrderConfirmation/_content.html.twig [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    {{ 'sylius.email.order_confirmation.your_order_number'|trans({}, null, localeCode) }}
+    <div style="margin: 10px 0;">
+      <span style="border: 1px solid #eee; padding: 10px; color: #1abb9c; font-size: 28px;">
+        {{ order.number }}
+      </span>
+    </div>
+    {{ 'sylius.email.order_confirmation.has_been_successfully_placed'|trans({}, null, localeCode) }}
+</div>
+
+
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/views/Email/Blocks/OrderConfirmation/_content.html.twig:11]:
```suggestion
```

File: src/Sylius/Bundle/CoreBundle/Resources/views/Email/layout.html.twig [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{% block body %}
+    {% autoescape false %}
+        {% set logo = channel.hostname is not null ? 'http://' ~ channel.hostname ~ asset('assets/shop/img/logo.png') : absolute_url(asset('assets/shop/img/logo.png')) %}
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/views/Email/layout.html.twig:3]: Logo is taken from shop namespace, I'm not sure if it is coupling CoreBundle with ShopBundle

File: src/Sylius/Bundle/CoreBundle/Tests/Mailer/OrderEmailManagerTest.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Test\Services\EmailChecker;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+final class OrderEmailManagerTest extends KernelTestCase
+{
+    public function testMailIsSentAndContentIsOk()
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Tests/Mailer/OrderEmailManagerTest.php:26]:
```suggestion
    public function testMailIsSentAndContentIsOk(): void
```

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Resources/views/Email/layout.html.twig (hunk 2 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/CoreBundle/Resources/views/Email/layout.html.twig:3]: It is just an asset. Perhaps we should move some image to CoreBundle as well, but for it should will allow the application to be run without Shop bundle.

**Comment by lchrusciel** [comment]
This PR extracted mailing from Shop to Core. This move allowed us to use the same email code and twig files in both Shop, Admin, and API context. Templates are moved, but in order to provide backward compatibility, files where they were used before, are left and include content from the core. To finish this PR, we should:
1. Refactor twig includes
1. Rebase 
1. Fix API Bundle composer.json
1. Consider refactoring mailing trigger to separate command (and perhaps introduce event bus, for integration layer) <- could be part of separate PR/task
1. Write ADR about placing emails & sending them from API context

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/composer.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
     "require": {
         "php": "^7.3",
         "api-platform/core": "^2.5",
-        "sylius/core-bundle": "^1.7",
+        "sylius/core-bundle": "dev-master",
```
[the remark this answers is not in the document]
Comment (lchrusciel) [reply src/Sylius/Bundle/ApiBundle/composer.json:28]: @SirDomin can you try to remove this line?

File: src/Sylius/Behat/Context/Setup/CheckoutContext.php [hunk 5]
```diff
@@ -64,6 +64,26 @@ public function __construct(
         $this->sharedStorage = $sharedStorage;
     }
 
+    /**
+     * @Given I have proceeded through checkout process in the :localeCode locale with email :email
+     */
+    public function iHaveProceededThroughCheckoutProcessInTheLocaleWithEmail(string $localeCode, string $email)
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Setup/CheckoutContext.php:70]:
```suggestion
    public function iHaveProceededThroughCheckoutProcessInTheLocaleWithEmail(string $localeCode, string $email): void
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Łukasz! :1st_place_medal:

**Pull request merged**